index.js

```diff
@@ -28,28 +29,39 @@ var COMPOSITION_RE_DIRECTIVE = /@compose (([A-Z][a-zA-Z]+,?[\s]*)+)/;
 
 function conformance(ast, reworkInstance) {
   var initialComment = ast.rules[0].comment;
-  var isDefinition = (initialComment && initialComment.match(/@define/));
+  var isApp = (initialComment && initialComment.match(/@app/));
   var isComponent = (initialComment && initialComment.match(RE_DIRECTIVE));
-  if (!isDefinition) { return; }
-  if (isDefinition && !isComponent) {
+  var isDefinition = (initialComment && initialComment.match(/@define/));
+  if (isApp && (isComponent || isDefinition)) {
```

Comment:
IDK if you'd prefer something like this:

```js
if (isDefinition) {
  if (isComponent) {
    // DO COMPONENT STUFF HERE!
  } else {
    // SHOW WARNINGS HERE
  }

  isApp = true;
}

if (isApp) {
  runBaseTests(rules);
}

```

IDK that the component/app warning is really super necessary. Up to you though!